db/migrate/20240501134003_add_data_fetched_to_planning_application_constraint.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddDataFetchedToPlanningApplicationConstraint < ActiveRecord::Migration[7.1]
+  def change
+    add_column :planning_application_constraints, :status, :string, default: "success", null: false
+
+    PlanningApplicationConstraint.find_each do |constraint|
+      status = constraint.data.nil? ? "failed" : "success"
```

Comment:
Not sure this is a valid check - I think it's an empty array by default:

https://github.com/unboxed/bops/blob/f70c5f465df1c87c213992ddca0c456dc376b947/engines/bops_api/app/jobs/bops_api/planning_application_dependency_job.rb#L55-L63

Maybe if it's populated mark it as `success` but `pending` otherwise?